src/main/java/org/humancellatlas/ingest/state/persistence/AutoLoader.java

```diff
@@ -29,23 +42,113 @@
     private final @NonNull SubmissionStateMonitor submissionStateMonitor;
     private final @NonNull Persister persister;
     private final @NonNull IngestApiClient ingestApiClient;
+    private final @NonNull StateMachineRepository<RedisRepositoryStateMachine> stateMachineRepository;
 
     private final Logger log = LoggerFactory.getLogger(getClass());
 
     public void autoLoad() {
```

Comment:
I think the way this file is organised makes it difficult to test. There is a lot of logic "hidden" in private functions. The only entry point is through the *persister*, which would make the testing cumbersome.

Reply:
Do you have a recommendation on how it should be organised?

Reply:
I've tried to extract the loading of a single state machine into a separate function so I wouldn't have to stub the persister.